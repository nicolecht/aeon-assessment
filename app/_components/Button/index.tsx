export default function Button({
  children,
  className,
  ...props
}: React.ButtonHTMLAttributes<HTMLButtonElement>) {
  return (
    <button
      className={`block p-3 font-semibold text-center text-white rounded-full min-w-32 bg-primary-500 hover:bg-primary-600 ${className}`}
      {...props}
    >
      {children}
    </button>
  );
}
