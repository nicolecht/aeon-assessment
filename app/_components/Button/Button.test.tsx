import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import Button from ".";

describe("Button component", () => {
  it("renders correctly and matches the snapshot", () => {
    const { asFragment } = render(<Button>Click Me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a button with the correct text", () => {
    render(<Button>Click Me</Button>);
    expect(
      screen.getByRole("button", { name: /click me/i })
    ).toBeInTheDocument();
  });

  it("calls onClick when clicked", async () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click Me</Button>);

    await userEvent.click(screen.getByRole("button", { name: /click me/i }));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
