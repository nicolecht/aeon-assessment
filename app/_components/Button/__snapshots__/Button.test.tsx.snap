// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="block p-3 font-semibold text-center text-white rounded-full min-w-32 bg-primary-500 hover:bg-primary-600 undefined"
  >
    Click Me
  </button>
</DocumentFragment>
`;
